src/block_cache/block_mgr.c

```diff
@@ -577,20 +579,24 @@ __bm_stat(WT_BM *bm, WT_SESSION_IMPL *session, WT_DSRC_STATS *stats)
 
 /*
  * __bm_switch_object --
- *     Switch the tiered object.
+ *     Switch the active handle to a different object.
  */
 static int
 __bm_switch_object(WT_BM *bm, WT_SESSION_IMPL *session, uint32_t objectid)
 {
     WT_BLOCK *block, *current;
     size_t root_addr_size;
 
+    /* The checkpoint lock protects against concurrent switches */
+    WT_ASSERT(session, FLD_ISSET(session->lock_flags, WT_SESSION_LOCKED_CHECKPOINT));
```

Comment:
Would it make sense to also add `WT_ASSERT_SPINLOCK_OWNED` here for the checkpoint lock, even though it would be redundant with the lock flags check?

Reply:
I'll switch it to the `SPINLOCK_OWNED` test. It seems better to be consistent with how we're now doing this elsewhere in the code.